client/email.go

```diff
@@ -0,0 +1,44 @@
+package client
+
+import (
+	"encoding/json"
+	"io/ioutil"
+
+	"github.com/hosting-de-labs/go-platform/model"
+)
+
+func (c *ApiClient) DomainSettingsFind(filter *RequestFilter) (*[]model.DomainSettingObject, error) {
+	currentPage := 0
+	var data []model.DomainSettingObject
+
+	for {
+		resp, err := c.runRequest("email", "domainSettingsFind", nil, 0, currentPage)
+		if err != nil {
+			return nil, err
+		}
+
+		defer resp.Body.Close()
```

Comment:
Don't use defer within a loop. Either define a function that is called or close the Body manually